circuits/src/zk_circuits/valid_commitments.rs

```diff
@@ -0,0 +1,950 @@
+
+        // Modify an order in the augmented wallet
+        witness.augmented_public_shares.orders[1].amount += Scalar::one();
+
+        assert!(!constraints_satisfied(witness, statement));
+    }
+
+    /// Test the case in which a prover attempts to modify a fee in the augmented wallet
+    #[test]
+    fn test_invalid_commitment__augmentation_modifies_fee() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (mut witness, statement) = create_witness_and_statement(&wallet);
+
+        // Modify a fee in the wallet
+        witness.augmented_public_shares.fees[0].gas_token_amount += Scalar::one();
+        assert!(!constraints_satisfied(witness, statement));
+    }
+
+    /// Test the case in which a prover attempts to modify wallet keys and blinders in augmentation
+    #[test]
+    fn test_invalid_commitment__augmentation_modifies_keys_blinder() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (original_witness, statement) = create_witness_and_statement(&wallet);
+
+        // Modify a key in the wallet
+        let mut witness = original_witness.clone();
+        witness.augmented_public_shares.keys.pk_match += Scalar::one();
+        assert!(!constraints_satisfied(witness, statement.clone()));
+
+        // Modify the wallet blinder
+        let mut witness = original_witness;
+        witness.augmented_public_shares.blinder += Scalar::one();
+        assert!(!constraints_satisfied(witness, statement))
+    }
+
+    /// Test the case in which the index of the send balance is incorrect
+    #[test]
+    fn test_invalid_commitment__invalid_send_index() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (witness, mut statement) = create_witness_and_statement(&wallet);
+
+        // Modify the index of the send balance
+        statement.balance_send_index += 1;
+
+        assert!(!constraints_satisfied(witness, statement))
+    }
+
+    /// Test the case in which the index of the receive balance is incorrect
+    #[test]
+    fn test_invalid_commitment__invalid_receive_index() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (witness, mut statement) = create_witness_and_statement(&wallet);
+
+        // Modify the index of the send balance
+        statement.balance_receive_index += 1;
+
+        assert!(!constraints_satisfied(witness, statement))
+    }
+
+    /// Test the case in which the index of the matched order is incorrect
+    #[test]
+    fn test_invalid_commitment__invalid_order_index() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (witness, mut statement) = create_witness_and_statement(&wallet);
+
+        // Modify the index of the order
+        statement.order_index += 1;
+
+        assert!(!constraints_satisfied(witness, statement))
+    }
+
+    /// Test the case in which a balance is missing from the wallet
+    #[test]
+    fn test_invalid_commitment__balances_missing() {
```

Comment:
super nit: should this be broken out into 3 tests?

Reply:
fixed